modules/linear_boltzmann_solvers/lbs_solver/io/flux_moments_io.cc

```diff
@@ -0,0 +1,248 @@
+// SPDX-FileCopyrightText: 2024 The OpenSn Authors <https://open-sn.github.io/opensn/>
+// SPDX-License-Identifier: MIT
+
+#include "modules/linear_boltzmann_solvers/lbs_solver/io/flux_moments_io.h"
+#include "modules/linear_boltzmann_solvers/lbs_solver/lbs_solver.h"
+#include "framework/mesh/mesh_continuum/mesh_continuum.h"
+
+namespace opensn
+{
+
+void
+WriteFluxMoments(LBSSolver& lbs_solver,
+                 const std::string& file_base,
+                 std::optional<const std::reference_wrapper<std::vector<double>>> opt_src)
+{
+  // Open file
+  std::string file_name = file_base + std::to_string(opensn::mpi_comm.rank()) + ".data";
+  std::ofstream file(file_name, std::ofstream::binary | std::ofstream::out | std::ofstream::trunc);
+  OpenSnLogicalErrorIf(not file.is_open(), "Failed to open " + file_name + ".");
+
+  std::vector<double>& src = opt_src.has_value() ? opt_src.value().get() : lbs_solver.PhiOldLocal();
```

Comment:
Two quick questions: 1) why use `PhiOldLocal` instead of `PhiNewLocal`, and 2) why use this `std::optional<std::reference_wrapper<std::vector<double>>> opt_dest` stuff instead of just requiring a vector? On 2), this eliminates hard-coding `PhiOldLocal`.

Reply:
I chose `PhiOldLocal` solely because that was the default here: `lua/modules/linear_bolzmann_solvers/lbs_solver/lbs_io_flux_moments.cc`.  I'm using `std::optional<std::reference_wrapper<std::vector<double>>> opt_dest` so that it's possible to optionally pass a reference to a vector.

Reply:
That's a fair point. Should we use `PhiNewLocal()` as the default? Is there any rationale to using `PhiOldLocal()`?